### ISSUE TITLE
False negative when evaluating arguments for metaclass `__call__` method in constructor call

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#9200]
The following should generate a type error, but it doesn't.

```python
from typing import Self, TypeVar

T = TypeVar("T")

class Meta1(type):
    def __call__(cls: type[T], x: int, y: str) -> T:
        return type.__call__(cls)

class Class1(metaclass=Meta1):
    def __new__(cls, *args, **kwargs) -> Self:
        return super().__new__(*args, **kwargs)

def func1(cls: type[Class1]):
    cls(x=1, y="")  # OK
    cls()  # E
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.385.
